### ISSUE TITLE
Missing ground light updates with Sildur 1283 (all presets)

### ISSUE THREAD

**Issue by MeeniMc** [IrisShaders/Iris#248]
# TL;DR

Skylight is not considered when rendering blocks that are 'under' another block. Note that this is different than having 'shadows', in the sense that even without shadows, lightlevels for blocks should be used to make blocks darker in low light conditions.

Possibly already reported in issue #153, but very unclear description then.

# How to replicate 

1. build a large floating platform
2. look under the platform
3. you will notice that mobs are 'dark', i.e., they are rendered with the appropriate light level
4. you will notice that blocks are 'light', i.e., they are **rendered as-if the plaform above had no effect on light level**
5. move under the platform
6. at first nothing will happen
7. after a couple of seconds, the scene will darken and everything will turn very dark (I think this is a feature of Sildur relating to dynamic aperture)

# Pictures a worth a million words?

### Mobs are dark, blocks are light, as-if lightlevels had been ignored
![2021-04-19_13 35 53](https://user-images.githubusercontent.com/68366846/115280883-67eefc80-a116-11eb-9cf8-7a584bbca99c.png)

### Moving under the platform, player gets dark, blocks remain light, until dynamic aperture (or whatever that Sildur feature is called) makes the whole scene dark
![2021-04-19_13 36 05](https://user-images.githubusercontent.com/68366846/115281308-db910980-a116-11eb-9d83-a84cbf057862.png)
![2021-04-19_13 36 06](https://user-images.githubusercontent.com/68366846/115281332-e0ee5400-a116-11eb-999b-9502fce29305.png)

# Versions

MC 1.16.5
Sildur 1.283 (any preset)

### IRIS:
```commit 0a72db99875908501abd7a60a15da110fff81447 (HEAD -> sodium-compatibility, origin/sodium-compatibility)
Merge: 27fd96c b769d98
Author: coderbot <coderbot16@gmail.com>
Date:   Fri Apr 9 17:32:17 2021 -0700

    Merge branch 'trunk' into sodium-compatibility
```

### SODIUM
```
commit c912291faf5eba01f47bb384f87b082648b994d9 (HEAD -> 1.16.x/iris)
Merge: 5027cbe 6caf15f
Author: MeeniMc <68366846+MeeniMc@users.noreply.github.com>
Date:   Mon Apr 19 12:31:48 2021 -0400

    Merge remote-tracking branch 'upstream/1.16.x/next' into 1.16.x/iris

commit 6caf15fcb84dcaa26175fe88d29cd9cd0e3ab6f3 (upstream/1.16.x/next)
Author: jellysquid3 <jellysquid3@users.noreply.github.com>
Date:   Sat Apr 17 13:16:58 2021 -0500

    change: Patch vanilla chunk manager instead of rolling our own

commit 5027cbe44542ab47e00b2fd2f672625acf340602 (origin/HEAD, origin/1.16.x/iris)
Merge: 6346945 026c2dc
Author: coderbot <coderbot16@gmail.com>
Date:   Mon Apr 12 11:44:25 2021 -0700

    Merge pull request #4 from DragonEggBedrockBreaking/1.16.x/iris
```
(Issue also present in prior, pure 1.16.x/iris, I just merged the latest Sodium, but it makes no difference)

# Hardware
RX580 win64 MD4.3

**Comment by solonovamax** [comment]
On the same commit with the same issue.

GPU: NVIDIA GeForce RTX 2070
GPU Driver: NVIDIA 465.24.02
OS: Arch Linux x84-64 5.11.15-arch1-2

So it's not windows specific, nor is it amd specific.

**Comment by solonovamax** [comment]
Actually, this is probably #84

**Comment by IMS212** [comment]
Try disabling shadows from the shaders.settings file, however this doesn't seem like the cause from the images.

**Comment by coderbot16** [comment]
This is a combination of #84 and #226 - entities turn darker much more quickly because the eye brightness is being updated much more quickly for entities than it should.

**Comment by MeeniMc** [comment]
Ok, the description of #226 clearly matches the delay observed for computing the overall brightness when under the platform. 

I do not think shadows (or lack thereof) explains what we see in the first image, since the vanilla renderer (which doesn't support shadows) will do the proper thing here, and use lightlevels to darken blocks as needed under the platform, even when the player is not standing under it. Maybe I am mistaken as to how Sildur handles lightlevels though.

**Comment by coderbot16** [comment]
Sildur's Vibrant handles sky lighting much more differently since it makes use of a shadow map. Areas where the vanilla light value is zero are treated as being in shadow (when the eye brightness is low) to prevent caves from being illuminated by the sun. Otherwise, things are treated as being in full brightness.

**Comment by thedocruby** [comment]
@MeeniMc I believe @coderbot16 is right in saying that this is linked to shadows: using the experimental shadows branch, this problem doesn't seem to occur. However, I believe that the second part of the issue where the screen goes black is directly linked to how Sildurs manages indoor lighting. When there are a sufficient amount of full opaque blocks between the player and the sky, (e.g. when in a cave or structure) Sildurs will stop rendering certain aspects of the game around the player, such as directional lighting (the reason the blocks underneath look sunlit) and volumetric lighting. This is to save a few frames where these features are not needed, e.g. in caves.
